### PR TITLE
Support for the --local flag in deployment-hooks

### PR DESCRIPTION
**Pull request by coreydaley**
Adds --local flag to allow for modifcations without communicating with the server api

Allows for overriding of deployment config options in file (using -f) by including the options on the command line

Fixes miscellaneous test errors in test/cmd/deployments.sh cause by missing oc create and oc delete commands

Fixes bug 1345773 - https://bugzilla.redhat.com/show_bug.cgi?id=1345773

### CODE REVIEW

**Comment by coreydaley** [comment]
[test]

**Comment by liggitt** [comment]
we should reach agreement upstream on how we want the `{oc,kubectl} set {noun}` commands to accomplish this, and do it consistently for all of them at once, rather than a one-off

File: test/cmd/deployments.sh [hunk 1]
```diff
@@ -145,6 +145,10 @@ os::test::junit::declare_suite_start "cmd/deployments/setdeploymenthook"
 arg="-f test/integration/testdata/test-deployment-config.yaml"
 os::cmd::expect_failure_and_text "oc set deployment-hook" "error: one or more deployment configs"
 os::cmd::expect_failure_and_text "oc set deployment-hook ${arg}" "error: you must specify one of --pre, --mid, or --post"
+os::cmd::expect_failure_and_text "oc set deployment-hook ${arg} -o yaml --pre -- mycmd" 'Error from server: deploymentconfigs'
+os::cmd::expect_success_and_text "oc set deployment-hook ${arg} --local -o yaml --post -- mycmd" 'mycmd'
+os::cmd::expect_success_and_not_text "oc set deployment-hook ${arg} --local -o yaml --post -- mycmd | oc set deployment-hook -f - --local -o yaml --post --remove" 'mycmd'
+os::cmd::expect_success 'oc create ${arg}'
```
Comment (stevekuznetsov) [inline test/cmd/deployments.sh]:
```sh
os::cmd::expect_success "oc create '${arg}'"
                        ^          ^      ^^
```

File: test/cmd/deployments.sh [hunk 2]
```diff
@@ -145,6 +145,10 @@ os::test::junit::declare_suite_start "cmd/deployments/setdeploymenthook"
 arg="-f test/integration/testdata/test-deployment-config.yaml"
 os::cmd::expect_failure_and_text "oc set deployment-hook" "error: one or more deployment configs"
 os::cmd::expect_failure_and_text "oc set deployment-hook ${arg}" "error: you must specify one of --pre, --mid, or --post"
+os::cmd::expect_failure_and_text "oc set deployment-hook ${arg} -o yaml --pre -- mycmd" 'Error from server: deploymentconfigs'
```
Comment (stevekuznetsov) [inline test/cmd/deployments.sh]:
```sh
os::cmd::expect_failure_and_text "oc set deployment-hook '${arg}' -o yaml --pre -- mycmd" 'Error from server: deploymentconfigs'
                                                         ^      ^
```

File: test/cmd/deployments.sh [hunk 3]
```diff
@@ -145,6 +145,10 @@ os::test::junit::declare_suite_start "cmd/deployments/setdeploymenthook"
...
+os::cmd::expect_success_and_text "oc set deployment-hook ${arg} --local -o yaml --post -- mycmd" 'mycmd'
```
Comment (stevekuznetsov) [inline test/cmd/deployments.sh]:
```sh
os::cmd::expect_success_and_text "oc set deployment-hook '${arg}' -o yaml --post -- mycmd" 'mycmd'
                                                         ^      ^
```

File: test/cmd/deployments.sh [hunk 4]
```diff
@@ -145,6 +145,10 @@ os::test::junit::declare_suite_start "cmd/deployments/setdeploymenthook"
...
+os::cmd::expect_success_and_not_text "oc set deployment-hook ${arg} --local -o yaml --post -- mycmd | oc set deployment-hook -f - --local -o yaml --post --remove" 'mycmd'
```
Comment (stevekuznetsov) [inline test/cmd/deployments.sh]:
```sh
os::cmd::expect_success_and_not_text "oc set deployment-hook '${arg}' -o yaml ...
                                                             ^      ^
```

**Comment by csrwng** [comment]
@liggitt this is bringing us to par with upstream. The --local flag was first introduced there. As we can, we need to implement it on all our set commands. This pull is working towards that. Per @deads2k request, I introduced a test to prevent additional set commands from not including it (https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/set/set_test.go).

**Comment by liggitt** [comment]
ok, cool... just wanted to make sure we were converging

**Comment by coreydaley** [comment]
@stevekuznetsov Updated syntax as requested (and discussed on irc)

**Comment by coreydaley** [comment]
@stevekuznetsov I have reverted my changes as per our chat

**Comment by coreydaley** [comment]
re[test]

**Comment by coreydaley** [comment]
@bparees Please review

**Comment by csrwng** [comment]
@coreydaley in your commit message... 
s/modifcations/modifications/

Also, squash your commits ... doesn't look like you need the second commit.
Other than that, LGTM

**Comment by csrwng** [comment]
One more thing... please remove deployment-hook from the exception list in this test:
https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/set/set_test.go

**Comment by coreydaley** [comment]
@csrwng  exception list modified, and commits squashed (not sure how I got two there, must have forgotten to force push)

**Comment by csrwng** [comment]
[test]

**Comment by coreydaley** [comment]
@csrwng Code has been updated

**Comment by coreydaley** [comment]
@csrwng please review and merge

**Comment by csrwng** [comment]
LGTM [merge]

**Pull request merged**